apps/src/sites/hourofcode.com/pages/public/index.js

```diff
@@ -1,20 +1,95 @@
-var google;
-var thanksUrl;
-var signupErrorMessage;
-var censusErrorMessage;
-var hocYear;
+/* globals google thanksUrl signupErrorMessage censusErrorMessage hocYear */
+
+import React from 'react';
+import ReactDOM from 'react-dom';
+import SchoolAutocompleteDropdown from '@cdo/apps/templates/SchoolAutocompleteDropdown.jsx';
+import SchoolNotFound from '@cdo/apps/templates/SchoolNotFound.jsx';
+
+var schoolData = {
+  nces: '',
+  schoolName: '',
+  schoolCity: '',
+  schoolState: '',
+  schoolZip: '',
+  schoolType: '',
+  showDropdownError: false,
+  showSchoolNotFoundError: false
+};
+
+function renderSchoolDropdown() {
+  ReactDOM.render (
+    <SchoolAutocompleteDropdown
+      setField={schoolDropdownOnChange}
+      value={schoolData.nces}
+      showErrorMsg={schoolData.showDropdownError}
+    />,
+    $('#school-selector')[0]
+  );
+}
+
+function renderSchoolNotFound() {
+  ReactDOM.render (
+    <SchoolNotFound
+      setField={schoolNotFoundOnChange}
+      schoolName={schoolData.schoolName}
+      schoolType={schoolData.schoolType}
+      schoolCity={schoolData.schoolCity}
+      schoolState={schoolData.schoolState}
+      schoolZip={schoolData.schoolZip}
+      showErrorMsg={schoolData.showSchoolNotFoundError}
+    />,
+    $('#school-not-found')[0]
+  );
+}
+
+function schoolDropdownOnChange(event) {
+  const val = (event ? event.value : '');
+  schoolData.nces = val;
+
+  if (!val) {
+    schoolData.showDropdownError = true;
+  } else {
+    schoolData.showDropdownError = false;
+  }
+
+  if (val === "-1"){
+    $('#school-not-found').show();
+  } else if (val){
+    $('#school-not-found').hide();
+  }
```

Comment:
It would be better if we had React own show/hide state instead of doing it in jquery. i.e. we have our outer component have state that is something like `this.state.showSchoolNotFound` and then in the render function we'd have something like
```jsx
render() {
  return (
    <div>
      {this.state.showSchoolNotFound && <SchoolNotFound .../>}
   </div>
}
```